daemon/graphdriver/overlay2/overlay.go

```diff
@@ -277,6 +294,17 @@ func (d *Driver) Create(id, parent, mountLabel string, storageOpt map[string]str
 		}
 	}()
 
+	if _, ok := storageOpt["size"]; ok {
+		driver := &Driver{}
+		if err := d.parseStorageOpt(storageOpt, driver); err != nil {
+			return err
+		}
+		// enforce container disk project quota
+		if err := d.quotaCtl.SetQuota(dir, driver.options.quota); err != nil {
```

Comment:
I guess you are right. I copied the parseStorageOpt() chunk from btrfs.go which does the same, but I can change it if you think it is important.